### PR TITLE
cli: rearrange versioning commands

### PR DESCRIPTION
**Pull request by pgavlin**
Reparent everything having to do with managing versions under the `esc env version` command and rearrange the command structure somewhat:

- `esc env version`: decribes a revision or tag
- `esc env version history`: show revision history
- `esc env version tag`: create or update a tagged revision
- `esc env version tag ls`: list tagged revisions
- `esc env version tag rm`: remove a tagged revision

### CODE REVIEW

File: cmd/esc/cli/env_diff.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func newEnvDiffCmd(env *envCommand) *cobra.Command {
 	diff := &envGetCommand{env: env}
 
 	cmd := &cobra.Command{
-		Use:   "diff [<org-name>/]<environment-name>[:<version>] [<version>]",
+		Use:   "diff [<org-name>/]<environment-name>[@<version>] [<version>]",
```
Comment (cleverguy25) [inline cmd/esc/cli/env_diff.go:26]: Didn't we land on:
```suggestion
		Use:   "diff [<org-name>/]<environment-name>[@<version>] [<environment-name][@<version>]",
```

File: cmd/esc/cli/env_version.go [hunk 2]
```diff
@@ -3,27 +3,126 @@
...
+	}
+
+	stamp := r.Created
+	if utc {
+		stamp = stamp.UTC()
+	} else {
+		stamp = stamp.Local()
+	}
+
+	fmt.Fprintf(stdout, "Date:   %v\n", stamp)
```
Comment (cleverguy25) [inline cmd/esc/cli/env_version.go:105]: Sorry if I am jumping the gun but the changelog said final.  We talked about outputting the tag metadata as well if the version passed in was a tag.

**Review by cleverguy25 — APPROVED** [review]

**Comment by cleverguy25** [comment]
Some nits from our conversation earlier but those could be a followup PR

**Comment by pgavlin** [comment]
> Some nits from our conversation earlier but those could be a followup PR

Yeah I didn't want to mix those changes in with this refactor. "final" in the CL just refers to the arrangement of the commands

**Pull request merged**